board.go

```diff
@@ -0,0 +1,69 @@
+package honeycombio
+
+// Compile-time proof of interface implementation.
+var _ Boards = (*boards)(nil)
+
+// Boards describes all the boards related methods that Honeycomb supports.
+type Boards interface {
+	// List all boards present in this dataset.
+	List() ([]Board, error)
+
+	// Get a board by its ID. Returns nil, ErrNotFound if there is no board
+	// with the given ID.
+	Get(id string) (*Board, error)
```

Comment:
We might want a `GetByName`?

Reply:
I would hold off a bit with adding new methods until there is a need for them. If this would simplify a downstream library, sure let's do it. But I want to avoid adding too many utility functions that we have to maintain but aren't used.